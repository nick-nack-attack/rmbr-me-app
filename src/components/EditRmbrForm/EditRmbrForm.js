import React, { Component } from 'react';
import { Textarea } from '../Utils/Utils';
import RmbrmeContext from "../../contexts/RmbrmeContext";
import RmbrApiService from "../../services/rmbr-api-service";
import { findRmbrByPersonId } from "../../helpers";

export default class EditRmbrForm extends Component {

    constructor(props) {
        super(props)
        this.state = {
            rmbrTitle: this.props.rmbr.rmbr_title,
            rmbrText: this.props.rmbr.rmbr_text,
            error: null,
            onCancelEdit: () => {},
            onHideEditForm: () => {}
        }
    };

    static contextType = RmbrmeContext;

    handleHideForm = event => {
        this.props.onHideEditForm()
    };

    handleTitleChange = event => {
        this.setState({
            rmbrTitle: event.target.value
        })
    };

    handleTextChange = event => {
        this.setState({
            rmbrText: event.target.value
        })
    }

    handleEditRmbrSubmit = e => {
        e.preventDefault()
        this.setState({error: null});
        const rmbr_title = this.state.rmbrTitle
        const rmbr_text = this.state.rmbrText
        const person_id = this.props.rmbr.person_id
        const user_id = window.localStorage.getItem('user_id')
        const rmbr_id = this.props.rmbr.id
        const category = this.props.rmbr.category
        const EditedRmbr = { rmbr_title, rmbr_text, person_id, user_id, category, rmbr_id };
        RmbrApiService.editRmbr(rmbr_id, EditedRmbr)
            .then(res => res.json())
            .then(data => {
                this.context.setRmbrArray(findRmbrByPersonId(data, person_id))
                this.handleHideForm()
            })
            .catch(err => {
                this.setState({
                    error:err.error
                })
            })
    };

    render() {
        // window.onload = document.getElementById('rmbrText').select();
        return (
            <form
                onSubmit={(e) => this.handleEditRmbrSubmit(e)}
            >
                <div>
                    <label>
                        Rmbr title
                    </label>
                    <Textarea
                        name='rmbrTitle'
                        id='rmbrTitle'
                        value={this.state.rmbrTitle}
                        onChange={this.handleTitleChange}
                        autoFocus={this.props.autofocus ? true : false}
                    />
                    <label>
                        Rmbr text
                    </label>
                    <Textarea
                        name='rmbrText'
                        id='rmbrText'
                        value={this.state.rmbrText}
                        placeholder='Add details here'
                        onChange={this.handleTextChange}
                        autoFocus={this.props.autofocus ? false : true}
                    />
                    <button>Save Changes</button>
                    <button
                        onClick={e => this.handleHideForm(e)}
                    >
                        Cancel
                    </button>
                </div>
            </form>
        )
    }

}