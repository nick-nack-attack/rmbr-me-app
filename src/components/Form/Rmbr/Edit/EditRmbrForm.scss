@import "../../../../themes/v2";

form {
    text-align: -webkit-center;
    &>legend {
        text-align: center;
    }
}