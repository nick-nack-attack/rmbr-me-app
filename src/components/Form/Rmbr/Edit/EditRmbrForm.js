// component to edit an existing rmbr
import React, { Component, useState, useContext } from 'react';

// context
import RmbrmeContext from "../../../../contexts/RmbrmeContext";
import { AppContext } from "../../../../contexts/AppContext";

// service
import AppApiService from "../../../../services/app-api-service";

// utils
import { Textarea, Button } from "../../../Utils/Utils";
import { findRmbrByPersonId } from "../../../Utils/helpers";

// style
import './EditRmbrForm.scss'

const EditRmbrForm = props => {

    const [ id, setId ] = useState(props.rmbr.id)
    const [ rmbr, setRmbr ] = useState(props.rmbr);
    const [ error, setError ] = useState(null);
    const context = useContext(AppContext);

    const handleHide = e => {
        e.preventDefault();
        props.onHideEditForm();
    };

    const handleSubmit = e => {
        e.preventDefault();
        const editedRmbr = { rmbr_title: rmbr.rmbr_title }
        AppApiService.editRmbr(id, editedRmbr)
            .then(res => {
                let returnedRmbr = res.find(rbr => rbr.id === id)
                props.onEditRmbrSuccess(returnedRmbr);
            })
    };

    return (
        <form onSubmit={ e => handleSubmit(e) }>
            <legend>Edit Rmbr</legend>
            <Textarea
                value={ rmbr.rmbr_title }
                onChange={ e => setRmbr({ rmbr_title: e.target.value }) }
                autoFocus
            />
            <Button
                label="Save"
                type="submit"
            />
            <Button
                label="Cancel"
                onClick={ e => handleHide(e) }
            />
        </form>
    );

};

export default EditRmbrForm;

/*

export default class EditRmbrForm extends Component {

    constructor(props) {
        super(props);
        this.state = {
            rmbrTitle: this.props.rmbr.rmbr_title,
            rmbrText: this.props.rmbr.rmbr_text,
            error: null,
            onCancelEdit: () => {},
            onHideEditForm: () => {}
        };
    };

    static contextType = RmbrmeContext;

    handleHideForm = event => {
        this.props.onHideEditForm();
    };

    handleTitleChange = event => {
        this.setState({
            rmbrTitle: event.target.value
        });
    };

    handleTextChange = event => {
        this.setState({
            rmbrText: event.target.value
        });
    };

    handleEditRmbrSubmit = e => {
        e.preventDefault();
        this.setState({error: null});
        const rmbr_title = this.state.rmbrTitle;
        const rmbr_text = this.state.rmbrText;
        const person_id = this.props.rmbr.person_id;
        const user_id = window.localStorage.getItem('user_id');
        const rmbr_id = this.props.rmbr.id;
        const EditedRmbr = { rmbr_title, rmbr_text, person_id, user_id, rmbr_id };
        AppApiService.editRmbr(rmbr_id, EditedRmbr)
            .then(res => res.json())
            .then(data => {
                this.context.setRmbrArray(findRmbrByPersonId(data, person_id));
                this.handleHideForm();
            })
            .catch(err => {
                this.setState({
                    error:err.error
                });
            })
    };

    render() {
        return (
            <form
                onSubmit={(e) => this.handleEditRmbrSubmit(e)}
            >
                <div>
                    <label>
                        Rmbr title
                    </label>
                    <Textarea
                        name='rmbrTitle'
                        id='rmbrTitle'
                        value={this.state.rmbrTitle}
                        onChange={this.handleTitleChange}
                        autoFocus={!this.props.autofocus}
                    />
                    <Button
                        label="Save"
                        className='edit_save_button edit_button'
                    />
                    <Button
                        label="Cancel"
                        className='edit_button'
                        onClick={e => this.handleHideForm(e)}
                    />
                </div>
            </form>
        );
    };
};

*/