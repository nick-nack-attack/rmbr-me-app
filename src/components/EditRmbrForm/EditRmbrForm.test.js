import React from "react";
import { shallow } from 'enzyme';
import toJson from "enzyme-to-json";
import EditRmbrForm from "./EditRmbrForm";

describe(`EditRmbrForm component`, () => {
    const rmbr = {
        id: 100,
        rmbr_title: 'Jack',
        rmbr_text: 'Friend',
        user_id: 1,
        person_id: 1,
        category: 'Current'
    }

    it(`renders the complete form`, () => {
        const wrapper = shallow(<EditRmbrForm/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    });
    it(`renders the title and content fields`, () => {
        const select = shallow(
            <EditRmbrForm
                rmbr={rmbr}
                autofocus={true}
            />
        )
            .find('#rmbrText', '#rmbrTitle')
        expect(toJson(select)).toMatchSnapshot()
    })
})