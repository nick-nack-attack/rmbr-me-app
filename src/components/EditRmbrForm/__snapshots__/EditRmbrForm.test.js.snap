// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditRmbrForm component renders the title and content fields 1`] = `
<Textarea
  autoFocus={false}
  id="rmbrText"
  name="rmbrText"
  onChange={[Function]}
  placeholder="Add details here"
  value="Friend"
/>
`;
